src/Dapr.PluggableComponents/Adaptors/DelegatedComponentProvider.cs

```diff
@@ -0,0 +1,23 @@
+using Grpc.Core;
+
+namespace Dapr.PluggableComponents.Adaptors;
+
+public sealed class DelegatedComponentProvider<TDelegatingType, TDelegatedType> : IDaprPluggableComponentProvider<TDelegatingType>
```

Comment:
It's not clear to me when we this is necessary. Would you mind to explain with an example? Thanks.

Reply:
For per-Dapr-instance multiplexing purposes, the gRPC adaptors defer fetching their `TComponentInterface` instance via a `IPluggableComponentProvider<TComponentInterface>` import (e.g. `IPluggableComponentProvider<IStateStore>`).  In cases where a single component implements several independently-adapted interfaces (e.g. both `IStateStore` and `IQueryableStateStore`), the other adaptors will expect component providers that match their expected interface types (e.g. `IPluggableComponentProvider<IQueryableStateStore>`.  In this example, however, as the instance management is already managed by the "main" `IPluggableComponentProvider<IStateStore>`, we need to delegate from an instance of `IPluggableComponentProvider<IQuerableStateStore>` to that existing `IPluggableComponentProvider<IStateStore>` instance.